### PR TITLE
feat: switch to `package-manager-detector`

### PR DESCRIPTION
**Pull request by benmccann**
### Description

Use `package-manager-detector` and remove the duplicated code from this package. I didn't re-export that functionality in favor of having folks use `package-manager-detector` directly, so this is a breaking change.

### CODE REVIEW

**Comment by antfu** [comment]
I would prefer to reexport it and make it non-breaking. We can look up the downstream usages and help them migrate first, before we finally drop those re-exports

**Comment by benmccann** [comment]
okay, thanks for taking a look. I've updated this to be non-breaking

**Comment by userquin** [comment]
Don't merge yet, I'm adding unknown package manager entry in detector

**Comment by userquin** [comment]
Released `detect-package-manager v0.1.1`

File: src/detect.ts [hunk 1]
```diff
@@ -6,5 +6,8 @@ export { type Agent } from 'package-manager-detector'
 export type PackageManager = 'pnpm' | 'yarn' | 'npm' | 'bun'
 
 export async function detectPackageManager(cwd = process.cwd()): Promise<Agent | null> {
-  return (await detect({ cwd }))?.agent || null
+  return (await detect({
+    cwd,
+    onUnknown: (packageManager) => { console.warn('[ni] Unknown packageManager:', packageManager) }
```
Comment (userquin) [inline src/detect.ts]: I guess this was a copy/paste from `ni` or viceversa, maybe it would be better to replace `[ni]` with `[@antfu/install-pkg]`

Comment (benmccann) [reply src/detect.ts]: done

File: package.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
     "lint": "eslint ."
   },
   "dependencies": {
-    "tinyexec": "^0.1.4"
```
Comment (userquin) [inline package.json]: uhmm, why is this dependency removed?

Comment (benmccann) [reply package.json]: oops. bad merge. fixed

**Pull request merged**